realm/realm-library/src/androidTest/java/io/realm/RealmObjectTests.java

```diff
@@ -1923,4 +1925,129 @@ public void execute(Realm realm) {
             }
         });
     }
+
+    @Test
+    public void getRealmConfiguration_managedRealmObject() {
+        realm.beginTransaction();
+        AllTypes object = realm.createObject(AllTypes.class);
+        realm.commitTransaction();
+
+        assertSame(realmConfig, RealmObject.getConfiguration(object));
+    }
+
+    @Test
+    public void getRealmConfiguration_managedRealmModel() {
+        realm.beginTransaction();
+        AllTypesRealmModel object = realm.createObject(AllTypesRealmModel.class, 1L);
+        realm.commitTransaction();
+
+        assertSame(realmConfig, RealmObject.getConfiguration(object));
+    }
+
+    @Test
+    public void getRealmConfiguration_DynamicRealmObject() {
+        final DynamicRealm dynamicRealm = DynamicRealm.getInstance(realmConfig);
+        //noinspection TryFinallyCanBeTryWithResources
+        try {
+            dynamicRealm.beginTransaction();
+            DynamicRealmObject object = dynamicRealm.createObject("AllTypesRealmModel", 1L);
+            dynamicRealm.commitTransaction();
+
+            assertSame(realmConfig, RealmObject.getConfiguration(object));
+        } finally {
+            dynamicRealm.close();
+        }
+    }
+
+    @Test
+    public void getRealmConfiguration_unmanagedRealmObjectThrows() {
+        try {
+            RealmObject.getConfiguration(new AllTypes());
+            fail();
+        } catch (IllegalArgumentException e) {
+            assertEquals(RealmObject.MSG_NOT_MANAGED_OBJECT, e.getMessage());
+        }
+    }
+
+    @Test
+    public void getRealmConfiguration_unmanagedRealmModelThrows() {
+        try {
+            RealmObject.getConfiguration(new AllTypesRealmModel());
+            fail();
+        } catch (IllegalArgumentException e) {
```

Comment:
Is this too harsh? Should we just return `null` in this case, e.g `isValid()` returns `true` for unmanaged objects instead of crashing for example?